### PR TITLE
Create CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by codereport**
Resolves https://github.com/thrust/thrust/issues/1133

### CODE REVIEW

**Comment by codereport** [comment]
If / once this is approved, I will duplicate the PR for `cub`.

**Comment by brycelelbach** [comment]
All reviewers please sign-off.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Contributor Covenant Code of Conduct
+
+## Overview
+
+Define the code of conduct followed and enforced for all RAPIDS projects.
```
Comment (griwes) [inline CODE_OF_CONDUCT.md:5]: We should probably change the RAPIDS reference to... whatever Bryce thinks is appropriate and will work for all the projects we may want to add this to.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## Scope
+
+This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at  [conduct@rapids.ai](mailto:conduct@rapids.ai)  All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
```
Comment (griwes) [inline CODE_OF_CONDUCT.md:45]: Same as above, we need a contact email address for Thrust here, not the RAPIDS one.

**Comment by brycelelbach** [comment]
Change it to cpp-conduct@nvidia.com

--------------------------------------------------------------------------------
Bryce Adelstein Lelbach aka wash
US Programming Language Standards (PL22) Chair
ISO C++ Library Evolution Chair
CppCon and C++Now Program Chair
CUDA Core C++ Libraries (Thrust, CUB, libcu++) Lead @ NVIDIA
--------------------------------------------------------------------------------

________________________________________

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (codereport) [reply CODE_OF_CONDUCT.md]: Done

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (codereport) [reply CODE_OF_CONDUCT.md]: Done

**Comment by allisonvacanti** [comment]
I can land this later today with some other changes that are going in.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+## Attribution
+
+This Code of Conduct was taken from the [NVIDIA RAPIDS](https://docs.rapids.ai/resources/conduct/) project, which was adapted from the  [Contributor Covenant](https://www.contributor-covenant.org/), version 1.4, available at https://www.contributor-covenant.org/version/1/4/code-of-conduct.html
+
+For answers to common questions about this code of conduct, see https://www.contributor-covenant.org/faq
+
+## Contact
+
+If you need to contact the Thrust team, please reach out to one of the following email adressess:
```
Comment (allisonvacanti) [inline CODE_OF_CONDUCT.md:57]: s/adressess/addresses/

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+- thrust-conduct@nvidia.com
+- cub-conduct@nvidia.com
+- libcudacxx-conduct@nvidia.com
```
Comment (allisonvacanti) [inline CODE_OF_CONDUCT.md:60]: Just checking -- do these email addresses exist? I only saw @brycelelbach mention `cpp-conduct@nvidia.com`.

Comment (codereport) [reply CODE_OF_CONDUCT.md:60]: He mentions the others in the issue: https://github.com/thrust/thrust/issues/1133

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (codereport) [reply CODE_OF_CONDUCT.md]: Done

**Comment by allisonvacanti** [comment]
This is ready to land (shelve 28408075) once I get a 👍 from @brycelelbach that the email accounts are active.

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# Contributor Covenant Code of Conduct
+
+## Overview
+
+Define the code of conduct followed and enforced for Thrust
+
+### Intended audience
+
+COMMUNITY | DEVELOPERS | PROJECT LEADS
```
Comment (brycelelbach) [inline CODE_OF_CONDUCT.md:9]: Can this be a bullet list instead?

**Comment by brycelelbach** [comment]
NVIDIA legal signed off. Please change the email to `cpp-conduct@nvidia.com` and then this is approved for merge.

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+This Code of Conduct was taken from the [NVIDIA RAPIDS](https://docs.rapids.ai/resources/conduct/) project, which was adapted from the  [Contributor Covenant](https://www.contributor-covenant.org/), version 1.4, available at https://www.contributor-covenant.org/version/1/4/code-of-conduct.html
+
+For answers to common questions about this code of conduct, see https://www.contributor-covenant.org/faq
+
+## Contact
+
+If you need to contact the Thrust team, please reach out to one of the following email addresses:
+- thrust-conduct@nvidia.com
+- cub-conduct@nvidia.com
+- libcudacxx-conduct@nvidia.com
```
Comment (brycelelbach) [inline CODE_OF_CONDUCT.md:62]: Change these to `cpp-conduct@nvidia.com`.

Comment (codereport) [reply CODE_OF_CONDUCT.md]: Done.

**Comment by brycelelbach** [comment]
@codereport can you rebase this? Looks like you have a conflict.

**Comment by brycelelbach** [comment]
Yes. Rebase commit would be best.

**Pull request merged**